torchdrivesim/simulator.py

```diff
@@ -1601,7 +1630,17 @@ def __init__(self, simulator: SimulatorInterface, res: Resolution = Resolution(1
 
         def record_birdview(simulator):
             s = simulator
-            bv = s.render(s.camera_xy, s.camera_psi, res=s.res, fov=s.fov)
+            waypoints = s.get_waypoints()
+            singleton = isinstance(s.agent_functor, SingletonAgentTypeFunctor)
+            if waypoints is not None:
+                waypoints_mask = s.get_waypoints_mask()
+                waypoints_dict = dict(agent=waypoints) if singleton else waypoints
+                waypoints_mask_dict = dict(agent=waypoints_mask) if singleton else waypoints_mask
+                waypoints = torch.cat(list(waypoints_dict.values()), dim=-3)
+                waypoints_mask = torch.cat(list(waypoints_mask_dict.values()), dim=-2)
```

Comment:
```suggestion
                waypoints = torch.cat(list(waypoints_dict.values()), dim=-3).flatten(1,2).unsqueeze(dim=1)
                waypoints_mask = torch.cat(list(waypoints_mask_dict.values()), dim=-2).flatten(1,2).unsqueeze(dim=1)
```

I think the above code should fix it but maybe render something first to make sure.